module/zfs/vdev_file.c

```diff
@@ -58,16 +51,31 @@ static taskq_t *vdev_file_taskq;
 static uint_t vdev_file_logical_ashift = SPA_MINBLOCKSHIFT;
 static uint_t vdev_file_physical_ashift = SPA_MINBLOCKSHIFT;
 
+void
+vdev_file_init(void)
+{
+	vdev_file_taskq = taskq_create("z_vdev_file", MAX(max_ncpus, 16),
```

Comment:
```
include/os/linux/spl/sys/sysmacros.h:#define    max_ncpus                       num_possible_cpus()
include/os/linux/spl/sys/sysmacros.h:#define    boot_ncpus                      num_online_cpus()
```
This has to do with systems (I think mostly VMs) where the number of CPUs can change. So a machine would be booted with support for 32 CPU cores, but only actually have 8 usable ones, and at some point in the future the other CPUs can be "onlined".

When scaling things, it is safer to scale based on the number of actually online CPUs, otherwise we might oversubscribe those CPUs.

You want to keep using `boot_ncpus` here instead of `max_ncpus`